### PR TITLE
mpfs: High resolution timer

### PR DESCRIPTION
**Pull request by jpaali**
Implements hrt for MPFS using MSTIMER of Polarfire SoC.

### CODE REVIEW

**Review by jlaitine — APPROVED** [review]
LGTM! Just put back the CMakeLists.txt and remove nuttx-udpate from the PR.
Then just merge

File: platforms/nuttx/src/px4/microchip/mpfs/hrt/hrt.c [hunk 1]
```diff
@@ -391,21 +503,26 @@ hrt_call_reschedule()
 		if (next->deadline <= (now + HRT_INTERVAL_MIN)) {
 			hrtinfo("pre-expired\n");
```
Comment (jlaitine) [inline platforms/nuttx/src/px4/microchip/mpfs/hrt/hrt.c:504]: Why not schedule here?

Comment (jpaali) [reply platforms/nuttx/src/px4/microchip/mpfs/hrt/hrt.c:504]: What do you mean? We set very short interval as it has been done in stm32 side.. but do you mean even shorter interval?
Or invoke() straight away?

Comment (jlaitine) [reply platforms/nuttx/src/px4/microchip/mpfs/hrt/hrt.c:504]: sorry, got confused with the diff; this is fine

**Pull request merged**